### ISSUE TITLE
[P2P] Redefine P2P concrete types in terms of interfaces

### ISSUE THREAD

**Issue by jessicadaugherty** [pokt-network/pocket#552]
## Objective

Redefine P2P concrete types in terms of interfaces between the P2P module and the underlying implementation.

## Origin Document

Next steps identified in https://github.com/pokt-network/pocket/pull/500

Part of https://github.com/pokt-network/pocket/issues/347

## Goals

- Improve modularity 
- Ensure that rainTreeNetwork does not depend on concrete p2p types

## Deliverable

- [ ] Update interface definitions for PeersManager, Peer, AddrBook, and AddrBookMap types respective methods and properties and remove any Raintree dependencies 
- [ ] Refactor code that uses concrete p2p types and update tests

## Non-goals / Non-deliverables

- Any additional raintree improvements or implementation 

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## Testing Methodology

- [ ] **All tests**: `make test_all`
- [ ] **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @jessicadaugherty

**Comment by jessicadaugherty** [comment]
Hey @bryanchriswhite @Olshansk can you review this ticket and confirm it should be pulled into the current iteration when you have a chance? Thank you

**Comment by bryanchriswhite** [comment]
@jessicadaugherty this looks great too me, thanks for putting it together! 🙌

I believe it's correct to move this to the current iteration because I'm currently working on it and it is the very next step after #347.

**Comment by jessicadaugherty** [comment]
Sorry for the delayed reply @bryanchriswhite - that sounds like a plan to me. Tysm!
